docs/user-guide/configuring/language-options.md

```diff
@@ -0,0 +1,214 @@
+* `mocha` - adds all of the Mocha testing global variables.
+* `jasmine` - adds all of the Jasmine testing global variables for version 1.3 and 2.0.
+* `jest` - Jest global variables.
+* `phantomjs` - PhantomJS global variables.
+* `protractor` - Protractor global variables.
+* `qunit` - QUnit global variables.
+* `jquery` - jQuery global variables.
+* `prototypejs` - Prototype.js global variables.
+* `shelljs` - ShellJS global variables.
+* `meteor` - Meteor global variables.
+* `mongo` - MongoDB global variables.
+* `applescript` - AppleScript global variables.
+* `nashorn` - Java 8 Nashorn global variables.
+* `serviceworker` - Service Worker global variables.
+* `atomtest` - Atom test helper globals.
+* `embertest` - Ember test helper globals.
+* `webextensions` - WebExtensions globals.
+* `greasemonkey` - GreaseMonkey globals.
+
+These environments are not mutually exclusive, so you can define more than one at a time.
+
+Environments can be specified inside of a file, in configuration files or using the `--env` [command line](https://eslint.org/docs/user-guide/command-line-interface) flag.
+
+### Using configuration comments
+
+To specify environments using a comment inside of your JavaScript file, use the following format:
+
+```js
+/* eslint-env node, mocha */
+```
+
+This enables Node.js and Mocha environments.
+
+### Using configuration files
+
+To specify environments in a configuration file, use the `env` key and specify which environments you want to enable by setting each to `true`. For example, the following enables the browser and Node.js environments:
+
+```json
+{
+    "env": {
+        "browser": true,
+        "node": true
+    }
+}
+```
+
+Or in a `package.json` file
+
+```json
+{
+    "name": "mypackage",
+    "version": "0.0.1",
+    "eslintConfig": {
+        "env": {
+            "browser": true,
+            "node": true
+        }
+    }
+}
+```
+
+And in YAML:
+
+```yaml
+---
+  env:
+    browser: true
+    node: true
+```
+
+### Using a plugin
+
+If you want to use an environment from a plugin, be sure to specify the plugin name in the `plugins` array and then use the unprefixed plugin name, followed by a slash, followed by the environment name. For example:
+
+```json
+{
+    "plugins": ["example"],
+    "env": {
+        "example/custom": true
+    }
+}
+```
+
+Or in a `package.json` file
+
+```json
+{
+    "name": "mypackage",
+    "version": "0.0.1",
+    "eslintConfig": {
+        "plugins": ["example"],
+        "env": {
+            "example/custom": true
+        }
+    }
+}
+```
+
+## Specifying Globals
+
+The [no-undef](https://eslint.org/docs/rules/no-undef) rule will warn on variables that are accessed but not defined within the same file. If you are using global variables inside of a file then it's worthwhile to define those globals so that ESLint will not warn about their usage. You can define global variables either using comments inside of a file or in the configuration file.
```

Comment:
There are a bunch of rules that rely on global variable information, so I wouldn't mention any specific rules here (it will be easy to forget to update this if things change.)

Reply:
I think I understand what you're saying here. To make the documentation reusable, it'll def be a good idea to stay away from specific rules here. I have a couple of suggestions on how to improve this paragraph, lmk what you think works well:

> ESLint has [rules] (https://eslint.org/docs/rules) that will warn if so some variables are accessed but not defined within the same file. If you are using global variables inside of a file then it's worthwhile to define those globals so that ESLint will not warn about their usage. You can define global variables either using comments inside of a file or in the configuration file.

> If you are using global variables inside of a file then it's worthwhile to define those globals so that ESLint will not warn about their usage. You can define global variables either using comments inside of a file or in the configuration file.

Reply:
My suggestion:

```suggestion
Some of ESLint's core rules rely on knowledge of the global variables available to your code at runtime. Since these can vary greatly between different environments as well as be modified at runtime, ESLint makes no assumptions about what global variables exist in your execution environment. If you would like to use rules that require knowledge of what global variables are available, you can define global variables in your configuration file or by using configure comments in your source code.
```

Reply:
I like mentioning to the related rules or `eslint:recommended`.

If someone configured globals, but ESLint doesn't warn about globals, they would be surprised. ESLint has many rules and it's hard to find related rules, so it's worthful that the "Specifying Globals" section describes how we configure warnings about globals. Probably `eslint:recommended` is suitable.

Reply:
We have a few different suggestions from team members here. I agree that it makes sense to avoid mentioning specific rules here for maintenance reasons, and don't feel strongly about how we go about solving that. Could we provide @khawarlatifkhan with a clear path forward so that we can land this?

Reply:
In the interest of moving things forward, let's go with @kaicataldo's suggestion here and we can always tweak the docs after it's merged.